crates/block-producer/src/mempool/batch_graph.rs

```diff
@@ -97,12 +100,18 @@ impl BatchGraph {
     ///   - any parent transactions are _not_ in the graph
     pub fn insert(
         &mut self,
-        transactions: Vec<TransactionId>,
+        transactions: Vec<(TransactionId, AccountId)>,
```

Comment:
Should we have a wrapper struct fro `(TransactionId, AccountId)` tuple? Not sure what to call it though. If we do introduce it, it should probably go into `miden-base`.